autowarefoundation/autoware
# Experimental/tsukuba challenge

## Status
**EXPERIMENTAL**

## Description
Customized Autoware to work with a small autonomous mobile robot and participated on Tsukuba Challenge 2017, and add our branch here for other roboticists to use and for the general Autoware community as well.

Different from a vehicle, a robot is very dynamic and thus several modifications were necessary on the planner, specially obstacle avoidance. Several changes to launch files and new drivers are also included. Found and fixed some issues on NDT localization. Minor changes on runtime manager to support persistence and resizable dialogs. The default parameters match our running configuration.

There is a big issue we could not fix on NDT_mapping estimating the correct height of a map, other robot teams confirmed the same.

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

Just run Autoware as always. Use the default parameters suit our robot needs but should be OK to fine tune things. The persistence on runtime manager will record some limited window state information to use at next run. 

This is the map generated (red) of the Tsukuba Challenge environment, the official course (yellow), our best autonomous run of over 1km (green) and the official trajectory of our expected participation.
![ndmap-w-google-map-all-routes-small](https://user-images.githubusercontent.com/16696954/34090982-1b95e7fa-e3fe-11e7-9809-f09be1f1f251.png)

This is a picture of the robot used, it has a Velodyne VLP-16, Hokuyo UTM-30LX, SH2 motor control box, Grasshopper camera, [Dynamic Vision Sensor (DVS)](https://inilabs.com/products/dynamic-vision-sensors/) event camera (using [rpg_dvs_ros](https://github.com/uzh-rpg/rpg_dvs_ros.git) drivers), no GPS/GNSS/IMU (lidar based localization). 
![robot-on-20171104-002-obstacle-avoid-2](https://user-images.githubusercontent.com/16696954/34091004-32ddbdf2-e3fe-11e7-8c4a-c07e17b9a1b7.png)

And here is OpenPlanner in action trying to avoid a large group of people.
![yyy-c](https://user-images.githubusercontent.com/16696954/34091022-57487114-e3fe-11e7-8ee6-555c4b4e347c.png)

Our rosbag and some maps and vector maps are stored on rosbag.tier4.jp (search for tsukuba)
